src/game_battler.cpp

```diff
@@ -221,19 +221,22 @@ bool Game_Battler::UseItem(int item_id) {
 			if (item->state_set.empty() || !item->state_set[0]) {
 				return false;
 			}
-
-			// Revive gives at least 1 Hp
-			if (hp_change == 0) {
-				ChangeHp(1);
-				was_used = true;
-			}
 		} else if (item->ko_only) {
 			// Must be dead
 			return false;
 		}
 
+		for (int i = 0; i < (int)item->state_set.size(); i++) {
+			if (item->state_set[i]) {
+				was_used |= HasState(Data::states[i].ID);
+				if (i == 0 && HasState(i + 1))
+					revived = 1;
+				RemoveState(Data::states[i].ID);
+			}
+		}
+
 		if (hp_change > 0 && !HasFullHp()) {
-			ChangeHp(hp_change);
+			ChangeHp(hp_change - revived);
```

Comment:
You always find something by looking long enough at it...
Any idea what this "- revived" is supposed to do here? It substracts 0 or 1? o.O